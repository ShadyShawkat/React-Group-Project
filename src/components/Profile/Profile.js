import React from 'react';
import { useSelector } from 'react-redux';

// import PropTypes from 'prop-types';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets.rockets.filter((rocket) => rocket.reserved));
  const missions = useSelector((state) => {
    const { missions } = state.missions;
    return missions.filter((missions) => missions.reserved);
  });
  return (
    <div className="flex justify-center gap-6">
      <div className="basis-1/3">
        <h3 className="text-xl font-bold">My Missions</h3>
        {missions.length === 0 && (
          <div className="text-3xl text-gray-400 font-bold select-none mt-3">
            Not member yet
          </div>
        )}
        {missions.length > 0 && (
          <ul className="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400">
            {missions.map((missions) => (
              <li
                data-testid="reservedMissionItem"
                key={missions.mission_id}
                className="p-2 flex gap-5 items-center"
              >
                <p>{missions.mission_name}</p>
              </li>
            ))}
          </ul>
        )}
      </div>
      <div className="basis-1/3">
        <h3 className="text-xl font-bold">My Rockets</h3>
        {rockets.length === 0 && (
          <div className="text-3xl text-gray-400 font-bold select-none mt-3">
            No rockets reserved
          </div>
        )}
        {rockets.length > 0 && (
          <ul className="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400">
            {rockets.map((rocket) => (
              <li
                key={rocket.id}
                className="p-2 flex gap-5 items-center"
                data-testid="reservedRocketItem"
              >
                <img
                  src={rocket.imageSrc}
                  className="w-8 h-8 rounded-sm"
                  alt="rocket"
                />
                <p>{rocket.name}</p>
              </li>
            ))}
          </ul>
        )}
      </div>
    </div>
  );
};

// Profile.propTypes = {};

// Profile.defaultProps = {};

export default Profile;
