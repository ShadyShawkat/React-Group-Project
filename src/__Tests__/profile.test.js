import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Profile from '../components/Profile/Profile';

const rocketsDummyData = [
  {
    rocket_id: '1',
    rocket_name: 'rocket 1',
    rocket_type: 'rocket',
    flickr_images: ['https://via.placeholder.com/150'],
    description: 'Rocket 1',
    reserved: false,
  },
  {
    rocket_id: '2',
    rocket_name: 'rocket 2',
    rocket_type: 'rocket',
    flickr_images: ['https://via.placeholder.com/150'],
    description: 'Rocket 2',
    reserved: true,
  },
  {
    rocket_id: '3',
    rocket_name: 'rocket 3',
    rocket_type: 'rocket',
    flickr_images: ['https://via.placeholder.com/150'],
    description: 'Rocket 3',
    reserved: true,
  },
];

const missionsDummyData = [
  {
    mission_name: 'Thaicom 1',
    mission_id: '9D1B7E0',
    description:
      'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    reserved: false,
  },
  {
    mission_name: 'Thaicom 2',
    mission_id: '1245sc',
    description:
      'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    reserved: true,
  },
  {
    mission_name: 'Thaicom 3',
    mission_id: 'vg12sc',
    description:
      'Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.',
    reserved: false,
  },
];

let tree = null;
describe('Tests Profile.js component render', () => {
  beforeEach(() => {
    const mockStore = configureStore();
    const store = mockStore({
      rockets: { rockets: rocketsDummyData },
      missions: { missions: missionsDummyData },
    });
    tree = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
  });
  it('Testing reserved rockets render', () => {
    expect(tree).toMatchSnapshot();
    const rocketItems = screen.getAllByTestId('reservedRocketItem');
    expect(rocketItems).toHaveLength(2);
  });
  it('Testing reserved missions render', () => {
    expect(tree).toMatchSnapshot();
    const missionItems = screen.getAllByTestId('reservedMissionItem');
    expect(missionItems).toHaveLength(1);
  });
});
