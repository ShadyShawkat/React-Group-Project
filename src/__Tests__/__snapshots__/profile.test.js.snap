// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Profile.js component render Testing reserved missions render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center gap-6"
      >
        <div
          class="basis-1/3"
        >
          <h3
            class="text-xl font-bold"
          >
            My Missions
          </h3>
          <ul
            class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
          >
            <li
              class="p-2 flex gap-5 items-center"
              data-testid="reservedMissionItem"
            >
              <p>
                Thaicom 2
              </p>
            </li>
          </ul>
        </div>
        <div
          class="basis-1/3"
        >
          <h3
            class="text-xl font-bold"
          >
            My Rockets
          </h3>
          <ul
            class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
          >
            <li
              class="p-2 flex gap-5 items-center"
              data-testid="reservedRocketItem"
            >
              <img
                alt="rocket"
                class="w-8 h-8 rounded-sm"
              />
              <p />
            </li>
            <li
              class="p-2 flex gap-5 items-center"
              data-testid="reservedRocketItem"
            >
              <img
                alt="rocket"
                class="w-8 h-8 rounded-sm"
              />
              <p />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center gap-6"
    >
      <div
        class="basis-1/3"
      >
        <h3
          class="text-xl font-bold"
        >
          My Missions
        </h3>
        <ul
          class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
        >
          <li
            class="p-2 flex gap-5 items-center"
            data-testid="reservedMissionItem"
          >
            <p>
              Thaicom 2
            </p>
          </li>
        </ul>
      </div>
      <div
        class="basis-1/3"
      >
        <h3
          class="text-xl font-bold"
        >
          My Rockets
        </h3>
        <ul
          class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
        >
          <li
            class="p-2 flex gap-5 items-center"
            data-testid="reservedRocketItem"
          >
            <img
              alt="rocket"
              class="w-8 h-8 rounded-sm"
            />
            <p />
          </li>
          <li
            class="p-2 flex gap-5 items-center"
            data-testid="reservedRocketItem"
          >
            <img
              alt="rocket"
              class="w-8 h-8 rounded-sm"
            />
            <p />
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests Profile.js component render Testing reserved rockets render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center gap-6"
      >
        <div
          class="basis-1/3"
        >
          <h3
            class="text-xl font-bold"
          >
            My Missions
          </h3>
          <ul
            class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
          >
            <li
              class="p-2 flex gap-5 items-center"
              data-testid="reservedMissionItem"
            >
              <p>
                Thaicom 2
              </p>
            </li>
          </ul>
        </div>
        <div
          class="basis-1/3"
        >
          <h3
            class="text-xl font-bold"
          >
            My Rockets
          </h3>
          <ul
            class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
          >
            <li
              class="p-2 flex gap-5 items-center"
              data-testid="reservedRocketItem"
            >
              <img
                alt="rocket"
                class="w-8 h-8 rounded-sm"
              />
              <p />
            </li>
            <li
              class="p-2 flex gap-5 items-center"
              data-testid="reservedRocketItem"
            >
              <img
                alt="rocket"
                class="w-8 h-8 rounded-sm"
              />
              <p />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center gap-6"
    >
      <div
        class="basis-1/3"
      >
        <h3
          class="text-xl font-bold"
        >
          My Missions
        </h3>
        <ul
          class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
        >
          <li
            class="p-2 flex gap-5 items-center"
            data-testid="reservedMissionItem"
          >
            <p>
              Thaicom 2
            </p>
          </li>
        </ul>
      </div>
      <div
        class="basis-1/3"
      >
        <h3
          class="text-xl font-bold"
        >
          My Rockets
        </h3>
        <ul
          class="flex flex-col mt-3 border-2 divide-y-2 divide-slate-300 rounded-md border-slate-400"
        >
          <li
            class="p-2 flex gap-5 items-center"
            data-testid="reservedRocketItem"
          >
            <img
              alt="rocket"
              class="w-8 h-8 rounded-sm"
            />
            <p />
          </li>
          <li
            class="p-2 flex gap-5 items-center"
            data-testid="reservedRocketItem"
          >
            <img
              alt="rocket"
              class="w-8 h-8 rounded-sm"
            />
            <p />
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
